plugin/src/main/java/org/opensearch/ml/action/prediction/TransportPredictionTaskAction.java

```diff
@@ -135,6 +139,8 @@ public void onResponse(MLModel mlModel) {
                                         } else {
                                             executePredict(mlPredictionTaskRequest, wrappedListener, modelId);
                                         }
+                                    } else if (functionName == FunctionName.REMOTE) {
+                                        deployAndPredictRemoteModel(modelId, wrappedListener, mlPredictionTaskRequest);
```

Comment:
Don't we need to check if the model is already deployed or not?

Reply:
It's checked in the function itself as the first thing.